diff --git a/android/src/main/java/expo/modules/medialibrary/MediaLibraryModule.kt b/android/src/main/java/expo/modules/medialibrary/MediaLibraryModule.kt
index dada864..c9f716b 100644
--- a/android/src/main/java/expo/modules/medialibrary/MediaLibraryModule.kt
+++ b/android/src/main/java/expo/modules/medialibrary/MediaLibraryModule.kt
@@ -110,7 +110,7 @@ class MediaLibraryModule : Module() {
     }
 
     AsyncFunction("createAssetAsync") Coroutine { localUri: String, albumId: String? ->
-      requireSystemPermissions()
+      // requireSystemPermissions()
       return@Coroutine createAssetWithAlbumId(context, localUri, true, albumId)
     }
 
